### PR TITLE
paganin filter

### PR DESCRIPTION
**Pull request by dkazanc**
faster implementation with some corrections of phase retrieval algorithm using Paganin filter from TomoPy.

### CODE REVIEW

File: httomolib/prep/phase.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    Returns
+    -------
+    np.ndarray
+        The 3D array of Paganin phase-filtered projection images.
+    """
+
+    # Check the input data is valid
+    if tomo.ndim == 2:
+        tomo = np.expand_dims(tomo, 0)
+
```
Comment (namannimmo10) [inline httomolib/prep/phase.py:74]: shall we only accept 3d data?
```suggestion
```

Comment (dkazanc) [reply httomolib/prep/phase.py:74]: we can do, yes.

File: httomolib/prep/phase.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    Args:
+        tomo (ndarray): 3d projection data
+
+    Returns:
+        ndarray: padded 3d projection data
+        tuple: a tuple with padding dimensions
+    """
+    full_shape_tomo = np.shape(tomo)
+        
+    pad_tup = np.zeros(np.array((3,2)), dtype=int)
```
Comment (namannimmo10) [inline httomolib/prep/phase.py:129]: this needs fixing!

**Comment by namannimmo10** [comment]
Like the commit messages! 😆

**Pull request merged**